client/client.go

```diff
@@ -70,28 +71,49 @@ func NewInsecureClient(rootURL *url.URL, timeout time.Duration) Client {
 }
 
 func newClient(rootURL *url.URL, timeout time.Duration, isInsecure bool) Client {
-	baseUrl := getBaseURL(*rootURL)
-
 	client := &client{
-		baseURL: baseUrl,
-		httpClient: &http.Client{
-			Timeout: timeout,
-			Transport: &http.Transport{
-				TLSClientConfig: &tls.Config{InsecureSkipVerify: isInsecure},
-			},
-		},
+		baseURL: getBaseURL(*rootURL),
+		httpClient: newHttpClient(timeout, isInsecure),
 	}
 	client.getApiLinks()
 	return client
 }
 
-
 // getBaseURL creates a url from the url path passed in and the apiVersion
 func getBaseURL(u url.URL) *url.URL {
 	u.Path = path.Join(u.Path, ApiVersion)
 	return &u
 }
 
+func newHttpClient(timeout time.Duration, isInsecure bool) *http.Client {
+	httpClient := &http.Client{
+		Timeout: timeout,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: isInsecure},
+		},
+	}
+
+	// this decorates the client transport with the jwt header
+	if config.GetJWT() != "" {
+		t := withHeader{Header: make(http.Header), rt: httpClient.Transport}
+		t.Set("Authorization", fmt.Sprintf("Bearer %s", config.GetJWT()))
+		httpClient.Transport = t
+	}
+	return httpClient
+}
+
+type withHeader struct {
+	http.Header
+	rt http.RoundTripper
+}
+
+func (h withHeader) RoundTrip(req *http.Request) (*http.Response, error) {
+	for k, v := range h.Header {
+		req.Header[k] = v
+	}
+	return h.rt.RoundTrip(req)
```

Comment:
rt could be nil I would either provide a constructor so it is more explicit that rt is required (I know you won't be able to force people to use it) or add check here and error or panic with meaningful message